examples/drawbuffers/main.js

```diff
@@ -0,0 +1,271 @@
+                'precision highp float;',
+                '#else',
+                'precision mediump float;',
+                '#endif',
+                'attribute vec3 Vertex;',
+                'attribute vec2 TexCoord0;',
+                'varying vec2 FragTexCoord0;',
+                'uniform mat4 ModelViewMatrix;',
+                'uniform mat4 ProjectionMatrix;',
+                'void main(void) {',
+                '  gl_Position = ProjectionMatrix * ModelViewMatrix * vec4(Vertex,1.0);',
+                '  FragTexCoord0 = TexCoord0;',
+                '}',
+                ''
+            ].join( '\n' );
+            var fragmentShader = [
+                '',
+                '#ifdef GL_ES',
+                'precision highp float;',
+                '#endif',
+                '#extension GL_EXT_draw_buffers : require',
+                'varying vec2 FragTexCoord0;',
+                'uniform sampler2D Texture0;',
+                'uniform sampler2D Texture1;',
+                'uniform sampler2D Texture2;',
+                'uniform sampler2D Texture3;',
+                'uniform sampler2D DepthTexture;',
+                'void main (void)',
+                '{',
+                '  vec3 color = texture2D( Texture0, FragTexCoord0.xy).xyz;',
+                '  gl_FragData[0] = vec4( color * texture2D( DepthTexture, FragTexCoord0.xy).xyz, 1.0 );',
+                '}',
+                ''
+            ].join( '\n' );
+            var program = new osg.Program( new osg.Shader( 'VERTEX_SHADER', vertexShader ), new osg.Shader( 'FRAGMENT_SHADER', fragmentShader ) );
+            return program;
+        },
+
+        createScene: function () {
+            var mrtGroup = new osg.Node();
+
+            var width = this._canvas.width;
+            var height = this._canvas.height;
+
+            var camera = this.createPreRenderCamera( width, height );
+            this.createRTTScene().then( function ( scene ) {
+                camera.addChild( scene );
+            } );
+
+            var ext = this._extDrawBuffers;
+            // createColorTextures
+            for ( var i = 0; i < NUM_TEXTURES; i++ ) {
+                this._textureList[ i ] = this.createColorTexture( width, height );
+                this._textureList[ i ]._name = 'texturecolor' + i;
+            }
+            var depthTexture = this.createDephTexture( width, height );
+            this._textureList.push( depthTexture );
+            // attach textures
+            camera.attachTexture( ext.COLOR_ATTACHMENT0_WEBGL, this._textureList[ 0 ], 0 );
+            camera.attachTexture( ext.COLOR_ATTACHMENT1_WEBGL, this._textureList[ 1 ], 0 );
+            camera.attachTexture( ext.COLOR_ATTACHMENT2_WEBGL, this._textureList[ 2 ], 0 );
+            camera.attachTexture( ext.COLOR_ATTACHMENT3_WEBGL, this._textureList[ 3 ], 0 );
+            camera.attachTexture( osg.FrameBufferObject.DEPTH_ATTACHMENT, depthTexture, 0 );
```

Comment:
Umm I don't get it, just tried to mimic a usual mrt setting in OSG https://github.com/openscenegraph/OpenSceneGraph/blob/master/examples/osgmultiplerendertargets/osgmultiplerendertargets.cpp#L349